pkg/cloudprovider/types.go

```diff
@@ -50,7 +51,7 @@ type CloudProvider interface {
 	// callback pattern to enable cloudproviders to batch capacity creation
 	// requests. The callback must be called with a theoretical node object that
 	// is fulfilled by the cloud providers capacity creation request.
-	Create(context.Context, *NodeRequest) (*v1.Node, error)
+	Create(context.Context, *v1alpha1.Machine) (*v1.Node, error)
```

Comment:
If there any impact on cpu/memory by computing these on the cloudprovider side. Should be able to check cloudprovider metrics.

Reply:
Sure, will check. I suspect that there will be a bit of a spike in the cloudprovider metrics, but the overall scheduling->launch time should be consistent since it's the same operations as before.

Reply:
It may be worthwhile to create a benchmarking suite for the cloudprovider calls now since we constantly find ourselves wanting to check what the timing looks like on these calls.

Reply:
Checked through Prom metrics. This PR (and the Cloudprovider one that it's paired with) have effectively no change to request time, scheduling time, memory, or CPU usage.